vortex-duckdb/src/exporter/dict.rs

```diff
@@ -67,6 +71,9 @@ impl<I: NativePType + AsPrimitive<u32>> ColumnExporter for DictExporter<I> {
         }
 
         vector.slice_to_dictionary(sel_vec, len);
+        // Use a unique id to each dictionary data array -- telling duckdb that the dict value vector
+        // is the same as reuse the hash in a join.
+        vector.set_dictionary_id(format!("{}-{}", self.cache_id, self.value_id));
```

Comment:
Maybe it's time to add the checksum statistic + compute fn to vortex? We _also_ want to know when two dictionaries are equal

Reply:
sounds like a good idea